mynn/layers/batchnorm.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+import mygrad as mg
+
+from mygrad.nnet.layers.batchnorm import batchnorm as tensor_batchnorm
+
+class batchnorm:
+    ''' A batch normalization layer.
+
+    This class will perform an n-dimensional batch normalization operation on an (N, D, ...)
+    shaped tensor scaled by γ of shape (D, ...) and shifted by β of shape (D, ...).
+    '''
+    def __init__(self, input_channels, momentum=0.1):
+        ''' Initialize a batch normalization layer.
+
+        Parameters
+        ----------
+        input_channels : int
+            The number of channels of the data to be batch-normalized.
+
+        momentum : float, optional (default=0.1)
+            The momentum value used to maintain moving averages.
+        '''
+        self.gamma = mg.ones((1, input_channels), dtype=np.float32)
+        self.beta = mg.zeros((1, input_channels), dtype=np.float32)
+
+        self.moving_mean = np.zeros((1, input_channels), dtype=np.float32)
+        self.moving_variance = np.zeros((1, input_channels), dtype=np.float32)
+    
+        self.momentum = momentum
+
+        self.input_channels = input_channels
+
+    def __call__(self, x, test=False):
+        ''' Perform the forward-pass of n-dimensional batch normalization over axis 0 on `x`.
+
+        Parameters
+        ----------
+        x : Union[numpy.ndarray, mygrad.Tensor], shape=(N, D, ...)
+            The data on which to perform batch normalization.
+
+        test : boolean, optional (default=False)
+            Determines whether the layer is being used at training time or test time.
+            This determines which mean and variance are used for normalization.
```

Comment:
I would add a little more exposition here than just saying "which mean and variance are used" for people who don't know what these different means are. Maybe just saying something like:

Whether to use averaged batch statistics or compute a mean and std over this batch